feat(nitrotype): init

MovByte opened the pull request:
## 🎉 Theme for [Nitro Type](https://nitrotype.com) 🎉

Nitro Type is a game that lets you improve your typing skills

## 💬 Additional Comments 💬

- This site requires an account to do much of anything, so you cannot reproduce what is in my catwalk screenshot

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - `catppuccin.user.css` - all the CSS for the userstyle, based on the
    template.
  - `preview.webp` - composite image of all four individual flavor screenshots stitched together,
    generated via [Catwalk](https://github.com/catppuccin/catwalk).

MovByte committed:
feat: add nitrotype port

MovByte committed:
fix(nitrotype): use the accent color for the "Race Results"

MovByte committed:
feat(nitrotype): support the race again tooltip

MovByte committed:
fix(nitrotype): only give the input elements that are selected the blue color

MovByte committed:
nitrotype(fix): apply color to the correct the switch in the friends switch

MovByte committed:
Apply suggestions from code review

To conform more to the catppuccin way of doing things and remove unused code

Co-authored-by: uncenter <47499684+uncenter@users.noreply.github.com>

MovByte committed:
Apply a suggestion from code review that for whatever reason went missing from the batch

Co-authored-by: uncenter <47499684+uncenter@users.noreply.github.com>

MovByte committed:
fix(nitrotype): fix licensing

MovByte committed:
feat(nitrotype): improve the input boxes for account login

MovByte committed:
style(nitrotype): use nesting whenever possible

MovByte committed:
nitrotype(feat): style the race box

MovByte committed:
feat(nitrotype): style the demo race

MovByte committed:
feat(nitrotype): improve the styling for the NT logo

MovByte committed:
feat(nitrotype): support the race results page and friend online notifications

MovByte committed:
feat(nitrotype): support the text color for the navbar

MovByte committed:
feat(nitrotype): support the text color for the dash actions

MovByte committed:
fix and feat(nitrotype): fix the text shadow (opaque) and borders, support the lootbox modal, and support more styling in the customizer

MovByte committed:
feat(nitrotype): support more dark shadow gradients using crust

MovByte committed:
chore: run linter

MovByte committed:
fix and feat(nitrotype): fix lint errors, keep gradients when possible, support dropdown links when hovered over, and improve friend online notifications

MovByte committed:
feat(nitrotype): add support for the warm up race box improve the race loader

MovByte committed:
feat(nitrotype): improve the race UI

MovByte committed:
fix(nitrotype): only the style the warm up race box when styleRaceBox is enabled

MovByte committed:
feat(nitrotype): various improvements

MovByte committed:
feat(nitrotype): style the race placement in the race results

MovByte committed:
feat(nitrotype): various improvements

MovByte committed:
feat(nitrotype): update styles

- update styles to account for UI updates
- switch to LESS with standard library

MovByte committed:
style(nitrotype): make better use of LESS features

MovByte committed:
feat(nitrotype): add new checkbox option `themeOnlyRaceBox`

MovByte committed:
feat(nitrotype): improvements for light flavors and style the Nitro Type logo on the homepage

uncenter commented on styles/nitrotype/preview.webp:
Can be deleted, we no longer store previews.

uncenter commented on styles/nitrotype/catppuccin.user.less:
```diff
@@ -0,0 +1,1739 @@
+@var select lightFlavor "Light Flavor" ["latte:Latte*", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha"]
+@var select darkFlavor "Dark Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
+@var select accentColor "Accent" ["rosewater:Rosewater", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve*", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Gray"]
+
+@var checkbox styleRaceBox "Style Race Box" 1
+@var checkbox themeOnlyRaceBox "Theme only the race box" 0
+@var checkbox preserveGoldGradient "Keep the gold lower gradients in boxes." 0
```
```suggestion
```

I don't see any reason we need to have options for only selectively theming these areas.

MovByte replied:
Can we keep those?
I find it useful to have the option for the theming just the race box, since your eyes are mostly focused on that when you are typing
It's where you spend most of your time while in a race

<img width="1065" height="841" alt="image" src="https://github.com/user-attachments/assets/36675337-026a-4d2e-a908-52372ed6d24a" />
<img width="1065" height="841" alt="image" src="https://github.com/user-attachments/assets/3597f552-9fd8-4c5b-89ca-a90d22c59874" />

I will remove them if you still insist

MovByte committed:
Update scripts/userstyles.yml

Co-authored-by: uncenter <uncenter@uncenter.dev>

MovByte commented on styles/nitrotype/catppuccin.user.less:
```diff
@@ -0,0 +1,1739 @@
+
+    @border: @crust;
+
+    /* the NitroType Gold coloring is kept because it is iconic */
+    @gold-start: #fadb66;
+    @gold-end: #fce797;
```
<img width="1065" height="819" alt="image" src="https://github.com/user-attachments/assets/bec767c3-8414-42f9-b537-1aa986e8225a" />

Should I make the many gold frames on the site Catppuccin Yellow colored?
Nitro Type Gold is the theme and literal name of the subscription they have, so the gold color is present in many places
Anytime a user's username shows up on the site, they also have this gold badge next to their name <img width="32" height="24" alt="image" src="https://github.com/user-attachments/assets/59bef7a0-c480-497e-ba3b-be0fe2652c28" />, for example

WalkQuackBack replied:
That could work, and you could apply the gradient effect between yellow and peach.

MovByte committed:
Update styles/nitrotype/catppuccin.user.less

Co-authored-by: uncenter <uncenter@uncenter.dev>

MovByte commented on styles/nitrotype/catppuccin.user.less:
```diff
@@ -0,0 +1,1739 @@
+/* '/teachers' path is disabled because it is unfinished */
+@-moz-document regexp("^https?://(www\\.)?nitrotype\\.com(?!/teachers).*") {
+  :root {
+    @media (prefers-color-scheme: light) {
```
> I'm a bit confused why both prefers-color-scheme and data-theme are being targeted, is this a leftover from the template?

Yes, I'll fix this
I started working on this theme over a year ago, and then I got busy

MovByte committed:
Update styles/nitrotype/catppuccin.user.less

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

MovByte commented on styles/nitrotype/catppuccin.user.less:
```diff
@@ -0,0 +1,1739 @@
+          color: @text;
+          &.waiting-for-leader--warm-up--word div[class*="letter-"].current {
+            background: @blue;
+            color: @text;
```
I understand where you're coming from, but let me show you a before (default theme) and after (current Catppuccin port)

<img width="517" height="198" alt="image" src="https://github.com/user-attachments/assets/7280a6e1-acb9-4ef1-91a4-470df4fafb9e" />
<img width="517" height="198" alt="image" src="https://github.com/user-attachments/assets/3e469587-3f67-4114-b15b-bd295df0b69e" />

I find it essential to be consistent with the UX of Nitro Type
This specifically is in the race box, where the user types out their sentences

MovByte replied:
Maybe I should make `colored-button-text` white in some cases

MovByte committed:
feat(nitrotype): fully style loading modal

MovByte committed:
fix(nitrotype): use `@text` var

MovByte committed:
fix(nitrotype): correct update URL of userscript

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

MovByte committed:
chore(nitrotype): remove theme preview

it is no longer needed

MovByte committed:
fix: use `border-color` to avoid redundancy

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

MovByte committed:
feat(nitrotype): enhance Catppuccin compliance and page support

- remove options to configure race box theming
- theme Gold properly
- make race box theming more faithful to original
- make text more visible
- correct text leader colors
- support teachers page

uncenter commented:
Closing as stale.
